common/http_test.cpp

```diff
@@ -1204,3 +1210,134 @@ TEST(HttpTest, TestTornDownStream) {
 	EXPECT_NE(error::NoError, err);
 	EXPECT_EQ(err.code, http::MakeError(http::StreamCancelledError, "").code);
 }
+
+TEST(HTTPSTest, CorrectSelfSignedCertificateSuccess) {
+	TestEventLoop loop;
+
+	bool client_hit_header {false};
+	bool client_hit_body {false};
+
+	mendertesting::TemporaryDirectory tmpdir;
+	string script = R"(#! /bin/sh
+	  exec openssl s_server -trace -www )";
+	script += " -key ../common/http/testdata/server.localhost.key";
+	script += " -cert ../common/http/testdata/server.localhost.crt";
+	script += " -accept " TEST_PORT;
+
+	const string script_fname = tmpdir.Path() + "/test-script.sh";
+	{
+		std::ofstream os(script_fname.c_str(), std::ios::out);
+		os << script;
+	}
+	int ret = chmod(script_fname.c_str(), S_IRUSR | S_IWUSR | S_IXUSR);
+	ASSERT_EQ(ret, 0);
+	processes::Process server({script_fname});
+	auto err = server.Start();
+	ASSERT_EQ(err, error::NoError);
+	std::this_thread::sleep_for(std::chrono::seconds {1}); // Give the server a little time to setup
```

Comment:
Tip: I just recently discovered this myself: You can write `std::this_thread::sleep_for(1s)` because `chrono` has a special handler for `s`, `ms` and the like!

Reply:
Sweet!